hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java

```
@@ -0,0 +1,85 @@
+ * limitations under the License.
+ */
+package com.hazelcast.internal.config.override;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.config.impl.YamlClientDomConfigProcessor;
+import com.hazelcast.config.Config;
+import com.hazelcast.internal.config.YamlMemberDomConfigProcessor;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import static com.hazelcast.internal.config.override.PropertiesToNodeConverter.propsToNode;
+import static java.lang.String.format;
+
+/**
+ * A class used to process external configuration overrides coming from environment variables/system properties,
+ * and applying them over existing {@link Config}/{@link ClientConfig} configuration
+ */
+public class ExternalConfigurationOverride {
+
+    private static final ILogger LOGGER = Logger.getLogger(ExternalConfigurationOverride.class);
+
+    public Config overwriteMemberConfig(Config config) {
+        return overwrite(config, (provider, c) -> {
+              try {
+                  new YamlMemberDomConfigProcessor(true, c)
+                    .buildConfig(new ConfigOverrideElementAdapter(propsToNode(provider.properties())));
+              } catch (Exception e) {
+                  throw new RuntimeException(e);
+              }
+          },
+          new EnvConfigProvider(EnvVariablesConfigParser.member()),
+          new SystemPropertiesConfigProvider(SystemPropertiesConfigParser.member()));
+    }
+
+    public ClientConfig overwriteClientConfig(ClientConfig config) {
+        return overwrite(config, (provider, c) -> {
+              try {
+                  new YamlClientDomConfigProcessor(true, c)
+                    .buildConfig(new ConfigOverrideElementAdapter(propsToNode(provider.properties())));
+              } catch (Exception e) {
+                  throw new RuntimeException(e);
```

`RuntimeException` seems a bit generic. Can we throw `InvalidConfigurationException` instead or something extending `HazelcastException`?